horace_core/sqw/PixelData/@PixelDataMemory/apply.m

```diff
@@ -0,0 +1,53 @@
+function [obj, data] = apply(obj, func_handle, args, data)
+% Apply a function (`func_handle`) to pixels (`obj`) with extra arguments `args`
+%  and recomputes the DnD if provided in `data`
+%
+% Inputs:
+%
+%   obj
+%        PixelDataFileBacked object
+%
+%   func_handle
+%        Function handle or cell array of function handles to apply
+%        `func_handle` must have a signature corresponding to:
+%
+%        pix_obj = func_handle(pix_obj, args{1}, ..., args{N})
+%
+%        N.B. `args` are the same for each function
+%
+%   args
+%        cell-array of extra args to pass to `func_handle`
+%
+%   data
+%        DnD object whose data are to be recomputed according to the
+%        result of `func_handle(obj,...)`
+
+    if ~exist('args', 'var') || isempty(args)
+        args = {{}};
+    end
+
+    if ~iscell(args)
+        args = {{args}};
+    end
+    if isa(func_handle, 'function_handle')
+        func_handle = {func_handle};
+    end
```

Comment:
excuse my paranoia but if func_handle is an array of function handles, do you need to expand it before making a cell of it

Reply:
If it's a cell array it will not get here. You cannot have an array of function handles. 
https://uk.mathworks.com/help/matlab/matlab_prog/fundamental-matlab-classes.html